.vsts.pipelines/phases/ci-linux.yml

```diff
@@ -0,0 +1,199 @@
+parameters:
+  dockerRegistryPassword: ''
+  dockerRegistryServer: ''
+  dockerRegistryUserName: ''
+  matrix: null
+  name: ci
+  queueDemands: Agent.OS -equals Linux
+  queueName: DotNet-Build
+
+phases:
+- phase: ${{ parameters.name }}
+  variables:
+    # Prefix to distinguish artifacts from different legs. No documented variable for this exists.
+    artifactName: no_artifact_name
+    buildLoggingOptions: ''
+    buildConfiguration: Release
+    buildOfflineTarball: false
+    # Use ":z" to set selinux flag for sharing in build-owned root dir. https://docs.docker.com/storage/bind-mounts/#configure-the-selinux-label
+    docker.map.agentSrc: -v $(Build.SourcesDirectory):/agentSrc -w /agentSrc
+    docker.map.drop: -v $(dropDirectory):/drop:z -w /drop
+    docker.map.logs: -v $(dropDirectory)/logs:/logs:z -w /logs
+    docker.map.root: -v $(rootDirectory):/root:z -w /root
+    docker.map.src: -v $(rootDirectory)/sb/source-build:/src:z -w /src
+    docker.map.tb: -v $(rootDirectory)/sb/tarball:/tb:z -w /tb
+    docker.run: docker run --rm
+    dockerRegistry.password: ${{ parameters.dockerRegistryPassword }}
+    dockerRegistry.server: ${{ parameters.dockerRegistryServer }}
+    dockerRegistry.userName: ${{ parameters.dockerRegistryUserName }}
+    dropDirectory: $(stagingDirectory)/drop
+    rootDirectory: $(Build.SourcesDirectory)/..
+    stagingDirectory: $(rootDirectory)/sb/staging
+    tarballName: tarball_$(Build.BuildId)
+  queue:
+    name: ${{ parameters.queueName }}
+    demands: ${{ parameters.queueDemands }}
+    timeoutInMinutes: 240
+    parallel: 2
+    matrix: ${{ parameters.matrix }}
+  steps:
+  - template: ../steps/docker-cleanup-linux.yml
+
+  # Docker registry login and pull, if one is defined.
+  - script: |
+      docker login -u $(dockerRegistry.userName) -p $(dockerRegistry.password) $(dockerRegistry.server)
+      docker pull $(imageName)
+    displayName: Docker login and image pull
+    condition: and(succeeded(), ne(variables['dockerRegistry.server'], ''))
+
+  # Docker registry logout, if one is defined. Do this immediately: avoid unnecessary login time.
+  - script: docker logout $(dockerRegistry.server)
+    displayName: Docker logout
+    condition: ne(variables['dockerRegistry.server'], '')
+
+  # Create working directory and copy source into it.
+  - script: |
+      set -x
+      $(docker.run) $(docker.map.root) $(docker.map.agentSrc) $(imageName) bash -c '
```

Comment:
It seems like this could be confusing regarding the working directory.  map.root and map.agentSrc both specify the -w option.  This happens in others, also.

Reply:
The last `-w` provided is used, but yeah, I don't know how to make that clear, and I think some of them don't even use `-w` (and some `-w`s aren't used). I'll change it to explicit variables.